tests/unit/test_step.py

```diff
@@ -359,29 +371,156 @@ def test_hierarchical_step_update_implementation_graph(
         assert edge in subgraph.edges(data=True)
 
 
-def test_loop_step() -> None:
-    step = STEP_KEYS["step_3"]
+@pytest.fixture
+def loop_step_params() -> Dict[str, Any]:
```

Comment:
Is this just a copy/paste of development.py? Seems like maybe we want to wrap each step in development.py in a getter function then so we're testing the actual code and don't need to worry about maintaining two things?

Reply:
> Is this just a copy/paste of development.py
 The pipeline schema doesn't have the additional secondary input slot that is being tested here. That is, I'm testing for functionality that isn't necessarily being utilized but needs to be possible.

I went back and forth about this a bit--one could just add the secondary slot to the "true" pipeline schema and reference that (that's what ```step = STEP_KEYS["step_3"]``` was doing before), but in the long term the non-dummy pipeline schema will probably not be using every capability that the particular steps it uses has, so I think it makes sense to test it explicitly.